pipelines/update_raster_stats.py

```diff
@@ -0,0 +1,148 @@
+from datetime import datetime
+
+import pandas as pd
+import xarray as xr
+from tqdm.auto import tqdm
+
+from src.constants import ISO3S
+from src.datasources import codab
+from src.utils import blob, database
+
+
+def get_existing_stats_dates(iso3: str, engine) -> list:
+    query = f"""
+    SELECT DISTINCT valid_date
+    FROM app.floodscan_exposure
+    WHERE iso3 = '{iso3.upper()}'
+    ORDER BY valid_date
+    """
+    df_unique_dates = pd.read_sql(query, con=engine)
+    df_unique_dates["valid_date"] = pd.to_datetime(
+        df_unique_dates["valid_date"]
+    )
+    return df_unique_dates["valid_date"].to_list()
+
+
+if __name__ == "__main__":
+    engine = database.get_engine(stage="dev")
+    clobber = False
+    verbose = False
+
+    for iso3 in ISO3S:
+        print(f"Processing {iso3}")
+        adm = codab.load_codab_from_blob(iso3, admin_level=2)
+        existing_exposure_rasters = [
+            x
+            for x in blob.list_container_blobs(
+                name_starts_with=f"{blob.PROJECT_PREFIX}/processed/"
+                f"flood_exposure/{iso3}/"
+            )
+            if x.endswith(".tif")
+        ]
+        existing_dates = get_existing_stats_dates(iso3, engine)
+        unprocessed_exposure_rasters = [
+            x
+            for x in existing_exposure_rasters
+            if datetime.strptime(x.split("/")[-1][13:23], "%Y-%m-%d")
+            not in existing_dates
+            or clobber
+        ]
+
+        # break list of exposure rasters into chunks, to avoid memory issues
+        # chunk length is arbitrary, but seems to work fine
+        chunk_len = 100
+        exposure_raster_chunks = [
+            unprocessed_exposure_rasters[x : x + chunk_len]
+            for x in range(0, len(unprocessed_exposure_rasters), chunk_len)
+        ]
+
+        # iterate over chunks
+        for exposure_raster_chunk in tqdm(exposure_raster_chunks):
+            # stack up exposure rasters in chunk
+            das = []
+            for blob_name in tqdm(exposure_raster_chunk):
+                date_in = datetime.strptime(
+                    blob_name.split("/")[-1][13:23], "%Y-%m-%d"
+                )
+                if date_in in existing_dates and not clobber:
```

Comment:
Is this redundant since we're already iterating through the `unprocessed_exposure_rasters`?

Reply:
Yes it is. This is just a leftover from a previous setup. d244751